sphinxcontrib/needs/needs.py

```diff
@@ -281,6 +284,7 @@ def setup(app):
     # doctree-read. So manipulating the doctree may result in conflicts, as e.g. images get not
     # registered for sphinx. So some sphinx-internal tasks/functions may be called by hand again...
     # See also https://github.com/sphinx-doc/sphinx/issues/7054#issuecomment-578019701 for an example
+    app.connect("doctree-resolved", load_needs_extra_filter_data)
```

Comment:
"doctree-resolved" gets executed per rst file.
So you load the data multiple times.

I would also use the function "load_config" to load the need_filter_data.
And then not by calling another function, but by simply:
```python

from sphinxcontrib/needs/filter_common.py  import  NEEDS_EXTERNAL_FILTER_DATA

def load_config():
...

global NEEDS_EXTERNAL_FILTER_DATA
NEEDS_EXTERNAL_FILTER_DATA = app.config['needs_filter_data']

```

Reply:
Done. This won't work, adapted to the approach as discussed.